sklearn/cross_decomposition/pls_.py

```diff
@@ -38,7 +39,9 @@ def _nipals_twoblocks_inner_loop(X, Y, mode="A", max_iter=500, tol=1e-06,
         # 1.1 Update u: the X weights
         if mode == "B":
             if X_pinv is None:
-                X_pinv = linalg.pinv(X)   # compute once pinv(X)
+                # We use slower pinv2 (same as np.linalg.pinv) for stability
+                # reason
```

Comment:
Quick benchmark. I think it is worth optimizing this, as it is trivial to do.
I would simply check where the execution stops when the input contains some `np.Inf`. We should just guarantee that an error is raised *before* calling `pinv2`. I suspect that we are already safe with the initial `check_array`.
```
>>> import numpy as np
>>> import scipy as sp

>>> X = np.random.RandomState(0).normal(0, 1, size=(1000, 5000))
>>> %timeit -n5 sp.linalg.pinv2(X)
5 loops, best of 3: 3.26 s per loop
>>> %timeit -n5 sp.linalg.pinv2(X, check_finite=False)
5 loops, best of 3: 3.03 s per loop

```